hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/qjournal/client/TestQuorumJournalManager.java

```diff
@@ -1101,6 +1113,59 @@ public void testSelectViaRpcTwoJNsError() throws Exception {
     }
   }
 
+  /**
+   * Test selecting EditLogInputStream after some journalNode jitter.
+   * And the corner case as below:
+   * 1. Journal 0 has some abnormal cases when journaling Edits with start txId 11.
+   * 2. NameNode just ignore the abnormal journal 0 and continue to write Edits to Journal 1 and 2.
+   * 3. Journal 0 backed to health.
+   * 4. Observer NameNode try to select EditLogInputStream vis PRC with start txId 21.
+   * 5. Journal 1 has some abnormal cases caused slow response.
+   *
+   * And the expected selecting result is: Response should contain 20 Edits from txId 21 to txId 40.
+   * Because there is no Edits from id 21 to 40 in the cache of JournalNode0.
+   */
+  @Test
+  public void testSelectViaRpcAfterJNJitter() throws Exception {
```

Comment:
This test passes even if I revert your production changes from `Journal`, can you check again?

Reply:
I think it's probably because you only make JN0 miss txns 1-10, but not 11-40, so I think it would throw `CacheMissException` right now.

Reply:
JN0 contains txns 1-10 and miss 11 ~ 40.